src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java

```diff
@@ -53,4 +151,121 @@ public OrderSummary getOrderSummary() {
+        HttpStatus httpStatus = Optional.ofNullable(HttpStatus.resolve(exception.status()))
+            .orElseGet(() -> HttpStatus.INTERNAL_SERVER_ERROR);
+
+        if (httpStatus == HttpStatus.NOT_FOUND) {
+            return new PbaResponse(httpStatus, String.format(NOT_FOUND.value(), pbaNumber), null);
+        }
+
+        CreditAccountPaymentResponse creditAccountPaymentResponse = getPaymentResponse(exception);
+
+        if (creditAccountPaymentResponse == null) {
+            return new PbaResponse(httpStatus, GENERAL.value(), null);
+        }
+
+        List<StatusHistoriesItem> statusHistories = Objects.requireNonNull(creditAccountPaymentResponse).getStatusHistories();
+
+        if (isEmpty(statusHistories)) {
+            return new PbaResponse(httpStatus, GENERAL.value(), null);
+        }
+
+        StatusHistoriesItem statusHistoriesItem = statusHistories.get(0);
+
+        String errorCode = statusHistoriesItem.getErrorCode();
+
+        String errorMessage = populateErrorResponse(pbaNumber, creditAccountPaymentResponse, httpStatus, errorCode);
+
+        return new PbaResponse(httpStatus, errorMessage, null);
+    }
+
+    private String populateErrorResponse(
+        String pbaNumber,
+        CreditAccountPaymentResponse creditAccountPaymentResponse,
+        HttpStatus httpStatus,
+        String errorCode
+    ) {
+        String errorMessage = null;
+        if (httpStatus == HttpStatus.FORBIDDEN) {
+            if (errorCode.equalsIgnoreCase(CA_E0001)) {
+                log.info("Payment Reference: {} Generating error message for {} error code",
+                    creditAccountPaymentResponse.getPaymentReference(),
+                    CA_E0001
+                );
+                errorMessage = String.format(CAE0001.value(), pbaNumber);
+            }
+            if (errorCode.equalsIgnoreCase(CA_E0004)) {
+                log.info("Payment Reference: {} Generating error message for {} error code",
+                    creditAccountPaymentResponse.getPaymentReference(),
+                    CA_E0004
+                );
+                errorMessage = String.format(CAE0004.value(), pbaNumber);
+            }
+
+        } else {
+            errorMessage = GENERAL.value();
+        }
+        return errorMessage;
+    }
+
+    private CreditAccountPaymentRequest creditAccountPaymentRequest(CaseData caseData, Long caseId) {
+        var creditAccountPaymentRequest = new CreditAccountPaymentRequest();
+        var orderSummary = caseData.getApplication().getApplicationFeeOrderSummary();
+
+        creditAccountPaymentRequest.setService(DIVORCE_SERVICE);
+        creditAccountPaymentRequest.setCurrency(GBP);
+        creditAccountPaymentRequest.setSiteId(caseData.getSelectedDivorceCentreSiteId());
+        creditAccountPaymentRequest.setAccountNumber(getPbaNumber(caseData));
+
+        final var solicitor = caseData.getApplicant1().getSolicitor();
```

Comment:
As far as the design of this class goes I think the public method needs to take two things: what the OrderSummary is (what are we paying for) and who is paying for it (app1/app2 solicitor). It's probably not good to hard code to app1 solicitor even if it's all we need now

Reply:
makes sense will pass solicitor as an argument to payment service method